contrib/mpi-proxy-split/split_process.cpp

```diff
@@ -333,7 +331,7 @@ startProxy()
       // Read from stdout of lh_proxy full lh_info struct, including orig memRange.
       close(pipefd_out[1]); // close write end of pipe
       // FIXME: should be a readall. Check for return error code.
-      if (read(pipefd_out[0], &lh_info, sizeof lh_info) < sizeof lh_info) {
+      if (read(pipefd_out[0], &lh_info, sizeof lh_info) < (ssize_t) sizeof lh_info) {
```

Comment:
Please do not commit the change o this line.

The issue here is that `read` returns an `int` and `sizeof` returns a `size_t`. (unsigned).  Some compiler complain or warn about comparing an `int` and `unsigned int`.  The type `ssize_t` stands for "signed size", which is almost the same thing as an `int`.  So, this cast is needed to make some compilers stop issuing warnings.